plugins/python/redisgears_python.c

```diff
@@ -1746,6 +1747,572 @@ static PyObject* atomicExit(PyObject *self, PyObject *args){
+
+static PyObject* PyDAG_ToStr(PyObject *obj){
+    return PyUnicode_FromString("PyDAGRunner to str");
+}
+
+static void PyDAGRunner_Destruct(PyObject *pyObj){
+    PyDAGRunner* pyDag = (PyDAGRunner*)pyObj;
+    if (pyDag->dag) RedisAI_DAGFree(pyDag->dag);
+    Py_TYPE(pyObj)->tp_free((PyObject*)pyObj);
+}
+
+static PyTypeObject PyDAGRunnerType = {
+  PyVarObject_HEAD_INIT(NULL, 0)
+  "redisgears.PyDAGRunner",             /* tp_name */
+  sizeof(PyDAGRunner), /* tp_basicsize */
+  0,                         /* tp_itemsize */
+  PyDAGRunner_Destruct,    /* tp_dealloc */
+  0,                         /* tp_print */
+  0,                         /* tp_getattr */
+  0,                         /* tp_setattr */
+  0,                         /* tp_compare */
+  0,                         /* tp_repr */
+  0,                         /* tp_as_number */
+  0,                         /* tp_as_sequence */
+  0,                         /* tp_as_mapping */
+  0,                         /* tp_hash */
+  0,                         /* tp_call */
+  PyDAG_ToStr,                         /* tp_str */
+  0,                         /* tp_getattro */
+  0,                         /* tp_setattro */
+  0,                         /* tp_as_buffer */
+  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,        /* tp_flags */
+  "PyDAGRunner",           /* tp_doc */
+};
+
+static PyObject* createDAGRunner(PyObject *cls) {
+    verifyRedisAILoaded();
+    RAI_DAGRunCtx* dagCtx = RedisAI_DAGRunCtxCreate();
+    PyDAGRunner* pyDag = PyObject_New(PyDAGRunner, &PyDAGRunnerType);
+    pyDag->dag = dagCtx;
+
+    return (PyObject*) pyDag;
+}
+/*
+ * Methods for RedisAI DAG runner
+ */
+
+static PyObject* loadInput(PyObject *self, PyObject *args) {
+    verifyRedisAILoaded();
+    if (RedisAI_DAGLoadTensor == NULL) {
+        PyErr_SetString(GearsError, "DAG run is not supported in RedisAI version");
+        return NULL;
+    }
+    PyDAGRunner* pyDag = (PyDAGRunner*)self;
+    if(!PyObject_IsInstance((PyObject*)pyDag, (PyObject*)&PyDAGRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type DAGRunner");
+        return NULL;
+    }
+    if(PyTuple_Size(args) != 2){
+        PyErr_SetString(GearsError, "Wrong number of args to DAG load input");
+        return NULL;
+    }
+    if(pyDag->dag == NULL){
+        PyErr_SetString(GearsError, "PyDAGRunner is invalid");
+        return NULL;
+    }
+    PyObject* tensorName = PyTuple_GetItem(args, 0);
+    if(!PyUnicode_Check(tensorName)){
+        PyErr_SetString(GearsError, "Tensor name argument must be a string");
+        return NULL;
+    }
+    const char* inputNameStr = PyUnicode_AsUTF8AndSize(tensorName, NULL);
+    PyTensor* pyt = (PyTensor*)PyTuple_GetItem(args, 1);
+    if(!PyObject_IsInstance((PyObject*)pyt, (PyObject*)&PyTensorType)){
+        PyErr_SetString(GearsError, "Given argument is not of type PyTensorType");
+        return NULL;
+    }
+    RedisAI_DAGLoadTensor(pyDag->dag, inputNameStr, pyt->t);
```

Comment:
Does this increase the tensor refcount or you need to create a shallow copy before?